src/System.Private.CoreLib/src/System/Threading/Thread.CoreRT.cs

```diff
@@ -34,6 +34,10 @@ public sealed partial class Thread
         // Protects starting the thread and setting its priority
         private Lock _lock;
 
+        private static readonly ManualResetEvent _allDone = new ManualResetEvent(false);
```

Comment:
Can this be initialized lazily only once somebody sets a foreground bit on a thread?

Reply:
I think we can do it. Please share your motivation, I guess we may want to avoid performance / resource penalty in the case there are no foreground threads created throughout program lifetime but I'm not sure that this is exactly what you suggesting. Also bear in mind that user threads are started in foreground mode unless ʻIsBackround` specified so we may been pursuing really rare opportunity for optimization.

Reply:
> I guess we may want to avoid performance / resource penalty in the case there are no foreground threads created throughout program lifetime 

Yes, exactly. My primary motivation is the extra code that it may keep for small programs.

Reply:
It is rare for user programs to start threads manually these days. Most people use Tasks.